### PR TITLE
Altering files

### PR DESCRIPTION
**Pull request by aarongerbrandt**
I altered some pretty cool files. Accidentally pulled a few extra changes in from a different branch but it was a lot more work to remove those I think.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,6 @@
 # CSC-303-Git-Playground
+### Back to main Class GitHub page [here](https://www.youtube.com/watch?v=dQw4w9WgXcQ)
```
Comment (SamBumgardner) [inline README.md:2]: Yes, this is correct.

File: run-me.py [hunk 2]
```diff
@@ -0,0 +1,2 @@
+import webbrowser
+webbrowser.open('https://www.youtube.com/watch?v=dQw4w9WgXcQ')
```
Comment (SamBumgardner) [inline run-me.py:2]: A joke so nice, we made it twice.

I hope nobody executes random python code on their machine without examining it first.

File: New Text Document.txt [hunk 3]
```diff
@@ -2,10 +2,10 @@
             /                                                    \
            |    _____________________________________________     |
            |   |                                             |    |
-           |   |  C:\User\Dude> java -verson                 |    |
-           |   |                                             |    |
-           |   |                                             |    |
-           |   |                                             |    |
+           |   |  C:\User\Dude> java -version                |    |
+           |   |  C:\User\Dude> echo $truth                  |    |
+           |   |  C# is Java but better.                     |    |
```
Comment (SamBumgardner) [inline New Text Document.txt:7]: This will surely be a spicy file for future students to discover. Good thinking.

**Review by SamBumgardner — CHANGES_REQUESTED** [review]
Changes look good inside this branch. 

It looks like you didn't check out master before making a new branch (you still had your "deleting files" branch checked out) which means that this branch includes all of your new changes + the stuff that was in your "deleting files" branch.

I think this should be relatively straightforward to fix with rebase (will involve running `git rebase master -i`) so let's sync up and get that fixed up if possible. 

If we end up with not enough time we could accept it as-is, but I think it's more valuable to understand how we can correct this sort of thing in the future.

**Comment by SamBumgardner** [comment]
Here's the specifics for how you can get those early commits removed from this branch *and* get this branch caught up with master at the same time.
1. if needed, do the "three step dance" to get master up-to-date everywhere: pull upstream master to your local master branch, then push local master to origin.
2. Checkout the branch you want to fix: 
    ```
    git checkout Altering_Files
    ```
3. Start a rebase to the `master` in interactive mode. This command lets us take some commits from this branch and re-apply them from a different starting point (we're "basing" them off of a different starting spot, in this case the current local master branch): 
    ```
    git rebase master -i
    ```
4. It should show you something like this terminal window:
image
5. We wanted this branch to start with commit `5e94ea2`, the one with the commit message "Buzzing! :)", right? In other words, we want to get rid of the first three commits.

    To do that, we can change the word `pick` in front of the first 3 commits to `drop` - that means when we complete the rebase, the `Altering_Files` branch will be scooted up to be based off of the current `master` branch, but it won't include those 3 commits. Example:
image
    If you have trouble navigating with vim, use `i` to enter 'insert mode' (you can edit text and navigate with arrow keys), press <esc> when you're done, then type `:wq` to exit vim and let git do its thing.
6. When finished, you should see a message like `Successfully rebased and updated refs/heads/Altering_Files.` If you run `git status, you'll see something like
    ``` 
    On branch Altering_Files
    Your branch and 'origin/Altering_Files' have diverged,
    and have 22 and 7 different commits each, respectively.
    ```
    At this point you can push up this fixed up thing as a new branch (`git push -u origin altering-files-fixed` or whatever) and make a fresh PR, or you can blow away the history of your origin/Altering_Files branch with `git push --force` - that'll force GitHub to overwrite its version of `Altering_Files` with the local version you're pushing up. If you force-push you won't have to make a new PR, the original PR will show the new commit history (which doesn't have those 3 delete commits) with a note of when you performed the force-push.

Feel free to reach out to me if anything goes wonky in this whole process. The big thing is, as always, don't push your git stuff out unless you're reasonably confident that the state of your branches and repository is how you want it to be.

**Comment by kellieb695** [comment]
I think I may have not checked out main before my 2nd branch as well, definitely going to refer back here when I need to clean things up/rebase my own branches. ~ Never gonna git you up ~

**Review by SamBumgardner — APPROVED** [review]
I think we're looking good now, history seems to be sorted out 👍

**Pull request merged**